internal/cephfs/store/volumeoptions.go

```diff
@@ -542,9 +542,10 @@ func (vo *VolumeOptions) populateVolumeOptionsFromBackingSnapshot(
 		return fmtBackingSnapshotOptionMismatch("clusterID", vo.ClusterID, parentBackingSnapVolOpts.ClusterID)
 	}
 
-	if vo.Pool != "" {
-		return errors.New("cannot set pool for snapshot-backed volume")
-	}
+	// Pool parameters is optional and only used to set 'pool_layout' arguement for
+	// subvolume and subvolume clone create commands.
+	// Setting this to emtpy since it is not used with Snapshot-backed volume.
+	vo.Pool = ""
```

Comment:
Add an E2E testing to ensure it works and we will not break it in the future.

Reply:
Done :+1: